### PR TITLE
Allow CLI path to have spaces

### PR DESCRIPTION
**Pull request by kartikgupta-db**

### CODE REVIEW

File: packages/databricks-vscode/src/configuration/ConnectionCommands.ts [hunk 1]
```diff
@@ -119,7 +119,7 @@ export class ConnectionCommands implements Disposable {
     }
 
     attachClusterQuickPickCommand() {
-        return async (title?: string) => {
+        return async () => {
```
Comment (ilia-db) [inline packages/databricks-vscode/src/configuration/ConnectionCommands.ts:122]: We still use the title in the `EnvironmentDependenciesVerifier->checkCluster`. We can remove the title from there too, but might as well leave it. Or is the `title` check here is not good enough and you are fixing a bug here by removing the argument?

File: packages/databricks-vscode/src/configuration/LoginWizard.ts [hunk 2]
```diff
@@ -384,25 +385,36 @@ export async function saveNewProfile(
         throw new Error("Can't save empty auth provider to a profile");
     }
 
-    const {path: configFilePath} = await loadConfigFile(
-        workspaceConfigs.databrickscfgLocation
-    );
+    const configFilePath: string = getDatabricksConfigFilePath().fsPath;
+    let shouldBackup = true;
+    try {
+        await loadConfigFile(configFilePath);
```
Comment (ilia-db) [inline packages/databricks-vscode/src/configuration/LoginWizard.ts:391]: Is the bug here in that we are failing to save the profile if the cfg file doesn't exist?

File: packages/databricks-vscode/src/cli/CliWrapper.ts [hunk 3]
```diff
@@ -19,9 +19,42 @@ import {quote} from "shell-quote";
...
-const execFile = promisify(execFileCb);
+function getEscapedCommandAndAgrs(
+    cmd: string,
+    args: string[],
+    options: SpawnOptionsWithoutStdio
+) {
+    if (process.platform === "win32") {
+        args = [
+            "/d",
+            "/c",
```
Comment (ilia-db) [inline packages/databricks-vscode/src/cli/CliWrapper.ts:33]: What do those magic args do? Can you add a link to some docs to the code?

File: packages/databricks-vscode/src/cli/CliWrapper.ts [hunk 4]
```diff
@@ -19,9 +19,42 @@ import {quote} from "shell-quote";
...
+            `""${cmd}" ${args.map((a) => `"${a}"`).join(" ")}"`,
+        ];
+        cmd = "cmd.exe";
```
Comment (ilia-db) [inline packages/databricks-vscode/src/cli/CliWrapper.ts:36]: Why are we changing the `cmd` here? In the `args` we will capture the old `cmd`, right? So it seems like this assignment should have any effects?

Comment (ilia-db) [reply packages/databricks-vscode/src/cli/CliWrapper.ts:36]: Oh wait, we return it later on, makes sense

File: packages/databricks-vscode/src/configuration/LoginWizard.ts (hunk 2 above)
Comment (kartikgupta-db) [reply packages/databricks-vscode/src/configuration/LoginWizard.ts:391]: Yep

File: packages/databricks-vscode/src/cli/CliWrapper.ts (hunk 4 above)
Comment (kartikgupta-db) [reply packages/databricks-vscode/src/cli/CliWrapper.ts:36]: We want to use cmd.exe to run this new command. Sort of like `bash -c "command"`

File: packages/databricks-vscode/src/configuration/ConnectionCommands.ts (hunk 1 above)
Comment (kartikgupta-db) [reply packages/databricks-vscode/src/configuration/ConnectionCommands.ts:122]: I messed up when fixing merge conflicts.

File: packages/databricks-vscode/src/bundle/BundleInitWizard.ts [hunk 5]
```diff
@@ -188,12 +188,12 @@ export class BundleInitWizard {
             "bundle",
             "init",
             "--output-dir",
-            this.cli.escapePathArgument(parentFolder.fsPath),
+            `"${parentFolder.fsPath}"`,
```
Comment (ilia-db) [inline packages/databricks-vscode/src/bundle/BundleInitWizard.ts:191]: I think mac and linux can technically have `"` in the dir names, so escaping the path can still be useful

**Review by ilia-db — APPROVED** [review]

**Pull request merged**